### PR TITLE
Add source and region code attribute support.

### PR DESCRIPTION
**Pull request by vicancy**
Follow up out-dated #963, resolving #910

### CODE REVIEW

File: test/Microsoft.DocAsCode.Metadata.ManagedReference.Tests/TripleSlashParserUnitTest.cs [hunk 1]
```diff
@@ -100,17 +121,19 @@ static int Main()
             {
                 AddReferenceDelegate = null,
                 PreserveRawInlineComments = false,
+                Source = new SourceDetail()
+                {
+                    Path = Path.Combine(inputFolder, "Source.cs"),
```
Comment (ansyral) [inline test/Microsoft.DocAsCode.Metadata.ManagedReference.Tests/TripleSlashParserUnitTest.cs]: >Path [](start = 20, length = 4)

what docfx generates is path relative to docfx.json? this way seems it doesn't work ?

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs [hunk 2]
```diff
@@ -258,6 +270,79 @@ private bool GetIsInheritDoc(XPathNavigator nav, ITripleSlashCommentParserContex
...
+                    return;
+                }
+
+                var region = node.Attribute("region");
+
+                var path = source.Value;
+                if (!Path.IsPathRooted(path))
+                {
+                    var directory = Path.GetDirectoryName(context.Source.Path);
+                    path = Path.Combine(context.Source?.BasePath ?? ".", directory, path);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: >context.Source?.BasePath ?? "." [](start = 40, length = 31)

if directory is path relative to docfx.json, File.Exists() doesn't work?

File: test/Microsoft.DocAsCode.Metadata.ManagedReference.Tests/TripleSlashParserUnitTest.cs [hunk 3]
```diff
@@ -3,10 +3,10 @@
...
 namespace Microsoft.DocAsCode.Metadata.ManagedReference.Tests
 {
-    using System.Collections.Generic;
-    using System.Linq;
-
+    using DataContracts.Common;
     using Microsoft.DocAsCode.DataContracts.ManagedReference;
+    using System.Collections.Generic;
+    using System.IO;
     using Xunit;
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Metadata.ManagedReference.Tests/TripleSlashParserUnitTest.cs]: sort

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs (hunk 2 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: Path is not relative to docfx.json, https://github.com/dotnet/docfx/blob/master/src/Microsoft.DocAsCode.Metadata.ManagedReference/Visitors/VisitorHelper.cs#L150, it is the local file path fetched from SyntaxTree

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs [hunk 4]
```diff
@@ -258,6 +270,80 @@ private bool GetIsInheritDoc(XPathNavigator nav, ITripleSlashCommentParserContex
             }
             return true;
         }
+
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: minor: remove the spaces?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**